optimum/exporters/onnx/model_configs.py

```diff
@@ -2641,6 +2642,25 @@ class EncoderDecoderOnnxConfig(EncoderDecoderBaseOnnxConfig):
     DEFAULT_ONNX_OPSET = 14  # uses SDPA in Transformers, hence opset>=14.
 
 
+class TimesFMOnnxConfig(OnnxConfig):
+    NORMALIZED_CONFIG_CLASS = NormalizedTimeSeriesForecastingConfig
+    MIN_TRANSFORMERS_VERSION = version.parse("4.52.0")
+    DUMMY_INPUT_GENERATOR_CLASSES = (TimesFMDummyInputGenerator,)
+    DEFAULT_ONNX_OPSET = 14  # uses SDPA in Transformers, needs opset>=14
+
+    @property
+    def inputs(self) -> Dict[str, Dict[int, str]]:
+        return {"past_values": {0: "batch_size", 1: "sequence_length"}}
```

Comment:
an issue is remaining before we can merge : it seems that the resulting onnx graph doesn't considered the `batch_size` as dynamic (won't accept one different than the one given during export) likely coming from :

https://github.com/huggingface/transformers/blob/79d4bc761d9da7a125b65a3b8c09441608d05436/src/transformers/models/timesfm/modeling_timesfm.py#L791

@kashif any chance we can add the modification directly in the transformers modeling ? otherwise we need to patch the model before export

Reply:
Indeed, that does look to be the source of the loss of dynamic shapes. torch.onnx.export doesn't play well with python list operations, so we'd need to refactor it to use tensor operations, if possible.